dts/bindings/misc/zephyr,flash-disk.yaml

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: Block storage device on flash partition
+
+compatible: "zephyr,flash-disk"
+
+include: base.yaml
+
+properties:
+    partition:
+      type: phandle
+      required: true
+      description: |
+        Backing storage flash map partition.
+
+    disk-name:
+      type: string
+      required: true
+      description: |
+        Disk name.
+
+    sector-size:
```

Comment:
Property name should be `block-size,` even previous Kconfig option was `DISK_FLASH_SECTOR_SIZE `, to make it clearer what side of the device it describes.

Reply:
@jfisher-no, I am doubting about this change of the property name. The block-size should be the allocation size, the sector-size should be either equal to the allocation size or less (e.g. 512 byte on a device that has a allocation size of 1kB, 2kB or 4kB). When this would be used in combination with nand-flash it would be needed to make a distinction and report the block-size different from the sector-size. The block-size could be picked up by the driver from the storage backend as the maximum of `sector-size` and `write-block-size`.